src/app/detail/ui/Sidebar.tsx

```diff
@@ -36,14 +43,15 @@ export const Sidebar = () => {
           <Selector
             getSelectedValue={(newValue) => {
               updateStatusValue(newValue)
+              // store.dispatch(setCreateTaskFields({ targetField: 'workflowStateId', value: (newValue as WorkflowStateResponse).id }))
             }}
-            startIcon={statusIcons[statusValue as StatusKey]}
-            options={status}
+            startIcon={statusIcons[(statusValue as WorkflowStateResponse).type]}
+            options={workflowStates}
             value={statusValue}
             selectorType={SelectorType.STATUS_SELECTOR}
             buttonContent={
               <Typography variant="bodySm" lineHeight="16px" sx={{ color: (theme) => theme.color.gray[600] }}>
-                {statusValue as ReactNode}
+                {(statusValue as WorkflowStateResponse).name as ReactNode}
```

Comment:
I see this throughout the code that we're using the `as` keyword in multiple places to treat the response as `WorkflowStateResponse`. Maybe we can use schema and a `parse` or just cast it to this type **at a top level** to avoid having to use `(xxx as WorkflowStateResponse)` in multiple places in the same file,

Reply:
Adds overall typecasting in this 7d6d402a798e1bd95743f3e548a2da6bc69ace03 commit